### DISCUSSION TITLE
Using 'choices_provider' Does Not Check Value(s)

### DISCUSSION THREAD

**Discussion by msmith626** [python-cmd2/cmd2#1192] [Q&A]
Hi,

Thank you for the wonderful software project! Using 'cmd2' for an interactive command-line application has been wonderful. I noticed that when using the 'choices_provider' argument with Cmd2ArgumentParser it does not validate the input given by the user, however, when using 'choices' it does validate (check input) the given value. I'm using version 2.2.0 and it appears the behavior is the same for master/tip, so I assume this is by design as only action.choices is considered in _ArgumentParser_check_value(). Using 'choices' is not an option for me since the choices permitted are dynamic (returned by a function, not a static list).

Should I just plan on checking values in my own code when using 'choices_provider' or is this a missing feature/enhancement that would be worthy to contribute?

Thanks,

Marc

**Comment by tleonhardt** [comment]
If you are using `choices_provider`, we expect that you will be doing your own validation.

It is potentially a missing feature/enhancement that would be useful, but I'm not sure if it could be done in a generic way. 

@kmvanbrunt @anselor Do either of you have any thoughts on this?
